pkg/operation/shoot/shoot.go

```diff
@@ -266,9 +266,14 @@ func (s *Shoot) GetNodeNetwork() gardencorev1alpha1.CIDR {
 	return *k8sNetworks.Nodes
 }
 
-// GetMachineImage returns the name of the used machine image.
-func (s *Shoot) GetMachineImage() *gardenv1beta1.ShootMachineImage {
-	return helper.GetMachineImageFromShoot(s.CloudProvider, s.Info)
+// GetCloudMachineImage returns the name of the used machine image.
+func (s *Shoot) GetCloudMachineImage() *gardenv1beta1.ShootMachineImage {
```

Comment:
```suggestion
func (s *Shoot) GetDefaultMachineImage() *gardenv1beta1.ShootMachineImage {
```

What I've understood is that if the `.spec.cloud.<provider>.machineImage` field is set then this value is used as default for all worker pools that don't explicitly configure a machine image themselves.

Reply:
Yes, correct. If .spec.cloud.<provider>.machineImage is not set, it comes with default machineImage.